apps/cpp_rpc/CMakeLists.txt

```diff
@@ -32,6 +32,14 @@ if (OS)
    endif()
 endif()
 
+if(USE_OPENCL)
+   if (ANDROID_ABI)
+     if($ENV{ANDROID_NDK_MAJOR} VERSION_LESS "23")
```

Comment:
I guess this is the problem - `$ENV`.

I don't think `$ENV` is needed. The CMake toolchain file from NDK should have already defined `ANDROID_NDK_MAJOR`. Hence, when that toolchain file is specified, `ANDROID_NDK_MAJOR` can be directly used as `${ANDROID_NDK_MAJOR}`. I think certain versions of CMake don't like `$ENV` being used with `if` and `VERSION_LESS`.